main_loop_tf/main.py

```diff
@@ -576,27 +643,48 @@ def build_graph(placeholders, input_shape, optimizer, weight_decay, loss_fn,
 
     # Convert from list of tensors to tensor, and average
     preds = tf.concat(tower_preds, axis=0)
+    s_preds = tf.concat(tower_preds[:smaller_n_gpus], axis=0)
     softmax_preds = tf.concat(tower_soft_preds, axis=0)
+    s_softmax_preds = tf.concat(tower_soft_preds[:smaller_n_gpus],
+                                axis=0)
 
     # Compute the mean IoU
     # TODO would it be better to use less precision here?
     mask = tf.ones_like(labels)
+    s_mask = tf.ones_like(smaller_labels)
```

Comment:
I suggest you also consider the case where the number of samples is a multiple of batch_size (i.e. where we don't need two graphs). You can do this before the for loop (and zip on out_list). 
```python
if n_small_gpus:
    out_vars = ([f_f_labels, s_labels], [f_preds, s_preds], [f_softmax_preds, s_softmax_preds])
else:
    out_vars = ([f_f_labels], [f_preds], [f_softmax_preds])
```

There might be a better name than out_vars for this. Pick the one you prefer.